### DISCUSSION TITLE
Update of tracked @AppStorage properties is not working when the View is wrapped in WithPerceptionTracking

### DISCUSSION THREAD

**Discussion by thomas-sempf** [pointfreeco/swift-perception#69] [Q&A]
### Description

The bug occurs in Views which are using WithPerceptionTracking and have one or more properties defined via @AppStorage. In this case a view update is not triggered when one of the @AppStorage properties changes.
[PerceptionBug.zip](https://github.com/pointfreeco/swift-perception/files/15184495/PerceptionBug.zip)

### Checklist

- [X] I have determined that this bug is not reproducible using Swift's observation tools. If the bug is reproducible using the `@Observable` macro or another tool from the `Observation` framework, please [file it directly with Apple](https://github.com/apple/swift/issues/new/choose).
- [X] If possible, I've reproduced the issue using the `main` branch of this package.
- [X] This issue hasn't been addressed in an [existing GitHub issue](https://github.com/pointfreeco/swift-perception/issues) or [discussion](https://github.com/pointfreeco/swift-perception/discussions).

### Expected behavior

View updates are still happening when @AppStorage properties changes

### Actual behavior

No update is triggered

### Steps to reproduce

1. Create a view with one or more @AppStorage properties and a @Perceptible view model
2. Wrap the view in WithPerceptionTracking { ... }
3. Find out that the updates, which should be propagated by the @AppStorage properties

### Perception version information

1.1.6

### Destination operating system

iOS 16

### Xcode version information

15.3

### Swift Compiler version information

```shell
swift-driver version: 1.90.11.1 Apple Swift version 5.10 (swiftlang-5.10.0.13 clang-1500.3.9.4)
Target: arm64-apple-macosx14.0
```

**Comment by stephencelis** [comment]
This is sadly a SwiftUI bug in which `@AppStorage` simply doesn't work in escaping closures. You'll see the same behavior if you swap `WithPerceptionTracking` out for a `GeometryReader`. To work around, you can move `@AppStorage` into its own dedicated subview.

We also recommend filing feedback with Apple if you'd like this bug to be fixed!

Since this is a bug with SwiftUI and not the library, I'm going to convert to a discussion.

**Reply by aehlke** [reply]
Do you happen to know if this is still an issue? cc @thomas-sempf 

With how common AppStorage usage is, this would be great to have as a disclaimer in the readme. Otherwise adopting this library without being aware of the SwiftUI bug is sure to introduce regressions in many projects which may not be easily caught by typical automated testing. Thanks
